python/pyspark/streaming/context.py

```diff
@@ -170,6 +173,48 @@ def getOrCreate(cls, checkpointPath, setupFunc):
         cls._transformerSerializer.ctx = sc
         return StreamingContext(sc, None, jssc)
 
+    @classmethod
+    def getActive(cls):
+        """
+        Return either the currently active StreamingContext (i.e., if there is a context started
+        but not stopped) or None.
+        """
+        activePythonContext = cls._activeContext
+        if activePythonContext is not None:
+            # Verify that the current running Java StreamingContext is active and is the same one
+            # backing the supposedly active Python context
+            activePythonContextJavaId = activePythonContext._jssc.ssc().hashCode()
+            activeJvmContextOption = activePythonContext._jvm.StreamingContext.getActive()
+            if activeJvmContextOption.isEmpty() or \
+                    activeJvmContextOption.get().hashCode() != activePythonContextJavaId:
```

Comment:
I think it's better to throw an exception when `activeJvmContextOption.get().hashCode() != activePythonContextJavaId` because it will be confusing that `getActive` returns None but `getActiveOrCreate` still complains there is already an active context in Scala codes.